pyanaconda/anaconda_argparse.py

```diff
@@ -404,6 +404,23 @@ def __call__(self, parser, namespace, values, option_string=None):
     # Network
     ap.add_argument("--proxy", metavar='PROXY_URL', help=help_parser.help_text("proxy"))
 
+    class SetWaitfornet(Action):
+        def __call__(self, parser, namespace, values, option_string=None):
+            value = None
+            try:
+                ivalue = int(values)
+            except ValueError:
+                pass
+            else:
+                if ivalue > 0:
+                    value = ivalue
+            if value is None:
+                value = 0
+            setattr(namespace, self.dest, value)
+
+    ap.add_argument("--waitfornet", dest="waitfornet", metavar="TIMEOUT_IN_SECONDS",
+                    action=SetWaitfornet, help=help_parser.help_text("waitfornet"))
+
```

Comment:
I am not quite sure about this, I've also tried to convert the type using type= argument but id didn't work for me.

`    def positive_int(value):
        rvalue = None.
        try:
            ivalue = int(value)
        except ValueError:
            pass
        else:
            if ivalue > 0:
                rvalue = ivalue
        if rvalue is None:
            rvalue = 0
        return rvalue

    ap.add_argument("--waitfornet", dest="waitfornet", metavar="TIMEOUT_IN_SECONDS",
                    default=0, type=positive_int, help=help_parser.help_text("waitfornet"))
`